test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -95,3 +510,154 @@ func cpuSpecToString(cpus *performancev2.CPU) string {
 	}
 	return sb.String()
 }
+
+// deleteTestPod removes guaranteed pod
+func deleteTestPod(testpod *corev1.Pod) {
+	// it possible that the pod already was deleted as part of the test, in this case we want to skip teardown
+	err := testclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(testpod), testpod)
+	if errors.IsNotFound(err) {
+		return
+	}
+
+	err = testclient.Client.Delete(context.TODO(), testpod)
+	Expect(err).ToNot(HaveOccurred())
+
+	err = pods.WaitForDeletion(testpod, pods.DefaultDeletionTimeout*time.Second)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// getOvnPod Get OVN Kubenode pods running on the worker cnf node
+func getOvnPod(workerNode *corev1.Node) (corev1.Pod, error) {
+	var ovnKubeNodePod corev1.Pod
+	ovnpods := &corev1.PodList{}
+	options := &client.ListOptions{
+		Namespace: "openshift-ovn-kubernetes",
+	}
+	err := testclient.Client.List(context.TODO(), ovnpods, options)
+	if err != nil {
+		return ovnKubeNodePod, err
+	}
+	for _, pod := range ovnpods.Items {
+		if strings.Contains(pod.Name, "node") && pod.Spec.NodeName == workerRTNode.Name {
+			ovnKubeNodePod = pod
+			break
+		}
+	}
+	testlog.Infof("ovn kube node pod running on %s is %s", workerNode.Name, ovnKubeNodePod.Name)
+	return ovnKubeNodePod, err
+}
+
+// getOvnPodContainers returns containerids of all containers running inside
+// ovn kube node pod
+func getOvnPodContainers(ovnKubeNodePod *corev1.Pod) ([]string, error) {
+	var ovnKubeNodePodContainerids []string
+	var err error
+	for _, ovnctn := range ovnKubeNodePod.Spec.Containers {
+		ctnName, err := pods.GetContainerIDByName(ovnKubeNodePod, ovnctn.Name)
+		if err != nil {
+			err = fmt.Errorf("unable to fetch container id of %v", ovnctn)
+		}
+		ovnKubeNodePodContainerids = append(ovnKubeNodePodContainerids, ctnName)
+	}
+	return ovnKubeNodePodContainerids, err
+}
+
+// getCpusUsedByOvnContainer returns cpus used by the ovn kube node container
+func getCpusUsedByOvnContainer(ovnKubeNodePodCtnid string) string {
+	var cpus string
+	var err error
+	var containerCgroup = ""
+	Eventually(func() string {
+		cmd := []string{"/bin/bash", "-c", fmt.Sprintf("find /rootfs/sys/fs/cgroup/cpuset/ -name '*%s*'", ovnKubeNodePodCtnid)}
```

Comment:
Again, we will have to rewrite all this in the next release unless we make it more generic.

Reply:
Addressed this in the latest commit

Reply:
@MarSik In this case  when i use the method of getting the cpus used by taskset,  i get only reserved cpus. Since the ovn kubenode pods contains a lot of containers, we don't have an api to execute commands on specific container in a pod.